plugin/src/main/java/com/playmonumenta/scriptedquests/zones/ZoneManager.java

```diff
@@ -22,12 +23,15 @@
 public class ZoneManager {
 	private Plugin mPlugin;
 	static BukkitRunnable mPlayerTracker = null;
+	static BukkitRunnable mAsyncReloadHandler = null;
 
 	private Map<String, ZoneLayer> mLayers = new HashMap<String, ZoneLayer>();
 	private Map<String, ZoneLayer> mPluginLayers = new HashMap<String, ZoneLayer>();
 	private ZoneTreeBase mZoneTree = null;
 	private Map<Player, ZoneFragment> mLastPlayerZoneFragment = new HashMap<Player, ZoneFragment>();
 	private Map<Player, Map<String, Zone>> mLastPlayerZones = new HashMap<Player, Map<String, Zone>>();
+	private Set<CommandSender> mReloadRequesters;
+	private Set<CommandSender> mQueuedReloadRequesters = new HashSet<CommandSender>();
```

Comment:
This needs to be a concurrent type or the plugin might crash as async and sync try to modify it

Reply:
I was wrong here too, this does work